mod/messageboard/actions/messageboard/delete.php

```diff
@@ -9,9 +9,7 @@
 $message = elgg_get_annotation_from_id($annotation_id);
 
 if ($message && $message->canEdit() && $message->delete()) {
-	system_message(elgg_echo("messageboard:deleted"));
-} else {
-	system_message(elgg_echo("messageboard:notdeleted"));
+	return elgg_ok_response('', elgg_echo('messageboard:deleted'));
```

Comment:
Not entirely sure if it will work after the delete. Maybe export to object and then pass as `deleted`?